### DISCUSSION TITLE
Filters web part, is showing filters for non-refinable managed property

### DISCUSSION THREAD

**Discussion by juliemturner** [microsoft-search/pnp-modern-search#1690] [Q&A]
First, I'm currently using version 4.5.3 of the Search Web Parts, but this saga begins with version 4.3.0.

When i started building this solution I was using search again people and trying to filter the results properly to make an employee directory but the client had specific requirements about showing the "About Me" formatted a certain way. During that time I created a filter on JobTitle and Department and that seemed to work well the only part they didn't like was formatting AboutMe.

We switch to a content type on a SharePoint list so that the AboutMe field could be a Rich Text field and could be formatted as needed. I created a development environment (which I have full access to Admin center) and added Search Results where I mapped the managed properties that the search indexer apparently automatically created that maps to the columns in the content type.  The managed properties were called EmployeeDepartmentOWSTEXT and EmployeeJobTitleOWSTEXT... they were retrievable, searchable, and queryable (but obviously not refinable).  They each are mapped to two crawled properties, as an example ows_EmployeeDepartment and ows_q_TEXT_EmployeeDepartment, I have no idea why there are two.  Ok, all is well in the world and I'm seeing just with the debug template the results showing up in the fields.  If I add a filter web part and then map to those managed properties (EmployeeDepartmentOWSTEXT and EmployeeJobTitleOWSTEXT) in my dev environment they actually come back with values.. I say actually because the reason I'm here is that this does not work in the clients environment and after reading apparently it shoudn't work in my dev environment at all because the managed properties are not refinable... but yet they do.  They return results and the filter works... however, the values are really weird.  So I'm just trying to figure out what this is even doing.  See screen shot;
note that the values are all lowercase and the spaces have been removed... but it does work... 

![image](https://user-images.githubusercontent.com/7570936/148260654-91772055-264e-483c-a551-3db003d90543.png)

**Comment by wobba** [comment]
You are using a non-refinable property as a refiner and there is an internal API bug in search where it actually gives you values back, and on the wrong form as you show. Hopefully this will be fixed soon from the Microsoft side, and you should stop getting values for non-refinable properties used as a refiner.

The logic used to show a list of properties cannot limit to just Refinable ones and maybe we get a schema API some time in the future which makes it easier to build better UX to avoid these issues.
